liquidctl/driver/smart_device.py

```diff
@@ -589,6 +592,9 @@ def _read_until(self, parsers):
     def _write_colors(self, cid, mode, colors, sval, direction='forward',):
         mval, mod3, mod4, mincolors, maxcolors = self._COLOR_MODES[mode]
 
+        if self._color_channels['sync'] == 0:
```

Comment:
With the change suggested before, this would become `if not self._color_channels:`